openapi-cli/src/main/java/io/ballerina/openapi/generators/openapi/OpenApiConverter.java

```diff
@@ -108,7 +115,8 @@ public void generateOAS3DefinitionsAllService(Path servicePath, Path outPath, St
                         } else {
                             content = definition.getYaml();
                         }
-                        CodegenUtils.writeFile(outPath.resolve(definition.getServiceName()), content.get());
+                        String fileName = checkDuplicateFiles(outPath, definition.getServiceName(), needJson);
```

Comment:
IMO using a method name like `resolveContractFileName` would be more meaningful. WDYT?

Reply:
Yeah, It would be fine.